src/main/java/htsjdk/samtools/SAMSequenceDictionaryCodec.java

```diff
@@ -31,14 +31,45 @@
  * "On the fly" codec SAMSequenceDictionaryCodec.
  * Encodes each sequence and directly writes it to the Dictionary file.
  *
+ * To use this class you should provide BufferedWriter to it, and so you should close it as you stop using this class.
+ * You can work with this class as shown below.
+ *
+ * Example of using this class:
+ *
+ * List<SAMSequenceRecord> dict = ...;
+ *
+ * //open BufferedReader and close in try-with-resources
+ * try(BufferedWriter writer = new BufferedWriter(new FileWriter("path/to/file"))) {
+ *      SAMSequenceDictionaryCodec codec = new SAMSequenceDictionaryCodec(writer);
+ *
+ *      //we have list of sequences, so encode header line and after that encode each sequence
+ *      codec.encodeHeaderLine(false);
+ *      dict.forEach(codec::encodeSequenceRecord);
+ *}
+ *
+ * or
+ *
+ * SAMSequenceDictionary dict = ...;
+ *
+ * //open BufferedReader and close in try-with-resources
+ * try(BufferedWriter writer = new BufferedWriter(new FileWriter("path/to/file"))) {
+ *      SAMSequenceDictionaryCodec codec = new SAMSequenceDictionaryCodec(writer);
+ *
+ *      //we have complete SAMSequenceDictionary, so just encode it.
+ *      codec.encode(dict);
```

Comment:
Method encode from `SAMSequenceDictionaryCodec` encodes the header, so we don't need to do it ourself:

```
public void encode(final SAMSequenceDictionary dictionary) { 
        encodeHeaderLine(false);
        dictionary.getSequences().forEach(this::encodeSequenceRecord);
}
```
As you have noticed, in the test I encode @HD line 2 times, I will fix it, or maybe it would be better to delete `encodeHeaderLine(false);` from `encode()` method and do it manually every time, when we use this class, what is your opinion?